Add Double Integrator Visualization

znewman01 opened the pull request:
Please take a look; fixes #209.

Note: I chose -20 < x < 20 for the X viewport: this is because the tolerance from the min-time cost function (0.05) makes the brick look off-center with lower limits. I chose the (q = -10, qdot = 0) starting conditions because they illustrate the "bang-bang" policy in this viewing region. (The policy seems to interpolate correctly even outside of the q in (-3, 3) range used for value iteration.)

Feel free to be nitpicky about code style; I didn't pay as much attention to surrounding code as I should have. Same for layout; I could easily move the DoubleIntegrator class into a double_integrator.py (with the DoubleIntegratorVisualizer); this resembles the structure in e.g. quadrotor2d.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/russtedrake/underactuated/212)
<!-- Reviewable:end -->

znewman01 commented:
(initial test failure was just being out-of-sync; a rebase fixed it)

RussTedrake commented on src/double_integrator/value_iteration.py:
```diff
@@ -109,6 +112,26 @@ def draw(iteration, mesh, cost_to_go, policy):
+builder = DiagramBuilder()
+plant = builder.AddSystem(DoubleIntegrator())
+logger = builder.AddSystem(SignalLogger(2))
+builder.Connect(plant.get_output_port(0), logger.get_input_port(0))
```
n.b. these two lines can be 
logger = LogOutput(plant.get_output_port(0), builder)

znewman01 replied:
Done.

RussTedrake commented on src/double_integrator/value_iteration.py:
```diff
@@ -109,6 +112,26 @@ def draw(iteration, mesh, cost_to_go, policy):
+
+diagram = builder.Build()
+simulator = Simulator(diagram)
+simulator.set_publish_every_time_step(False)
```
you can zap this line (it's no longer needed)

znewman01 replied:
Done.

RussTedrake commented on src/double_integrator/value_iteration.py:
```diff
@@ -109,6 +112,26 @@ def draw(iteration, mesh, cost_to_go, policy):
+simulator.set_publish_every_time_step(False)
+
+state = simulator.get_mutable_context().get_mutable_continuous_state_vector()
+state.SetFromVector([-10.0, 0.0])
```
These two lines can now be just
simulator.get_mutable_context().SetContinuousState([-10.0, 0.0])

RussTedrake commented on src/double_integrator/visualizer.py:
```diff
@@ -0,0 +1,62 @@
+import numpy as np
+import matplotlib.pyplot as plt
```
can you add visualizer.py to the CMakeList.txt like this:
https://github.com/RussTedrake/underactuated/blob/master/src/quadrotor2d/CMakeLists.txt#L2

(i think it will then ask you to change a little of the formatting)

znewman01 replied:
Done (no format changes requested, apparently)

RussTedrake commented on src/double_integrator/visualizer.py:
```diff
@@ -0,0 +1,62 @@
+
+    def __init__(self):
+        PyPlotVisualizer.__init__(self)
+        self.DeclareInputPort(PortDataType.kVectorValued, 1)
```
should have size 2 now?

znewman01 replied:
Good catch, done.

znewman01 commented:
Alright, ready for another look!

(I think I clobbered the review history by force-pushing; I'm used to code review tools that where that's the workflow and haven't used GitHub in years, oops. The only changes I made were the ones directly addressed your comments.)
